components/DashboardProfile/index.js

```diff
@@ -10,33 +10,57 @@ class DashboardProfile extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
+      disabled: false,
       edit: false,
       user_metadata: {
         imagePreviewUrl: this.props.profile.imagePreviewUrl || '',
-        nickname: this.props.profile.nickname || ''
+        nickname: this.props.profile.nickname || '',
+        userEmail: this.props.profile.userEmail || ''
       }
     };
   }
   handleEdit = () => {
     this.setState({ edit: true });
   };
+  validateEmail(email) {
+    var re = /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+    return re.test(email);
+  }
   handleChangeUserMetaData = (field, e) => {
-    let value;
-    if (!e.target.value) {
-      value = this.props.profile.nickname;
+    if (field === 'userEmail') {
+      if (e.target.value === '') {
+        this.setState({ disabled: false });
+      } else {
+        if (this.validateEmail(e.target.value)) {
+          this.setState({
+            disabled: false,
+            user_metadata: {
+              ...this.state.user_metadata,
+              userEmail: e.target.value
+            }
+          });
+        } else {
+          this.setState({ disabled: true });
+          swal(
+            'Please enter a valid email address or leave empty',
+            null,
+            'error'
+          );
+        }
+      }
```

Comment:
extract this out into a `if(e.target.value != '') { switch(userField) { case 'userEmail':  } }

Reply:
could also set the message you pass to sweetalert in a variable and just change it in the else statements of the different fields